gems/sorbet-runtime/lib/types/_types.rb

```diff
@@ -88,8 +88,12 @@ def self.class_of(klass)
   #
   # The name of the type alias is not preserved; Error messages will
   # be printed with reference to the underlying type.
-  def self.type_alias(type)
-    T::Utils.coerce(type)
+  def self.type_alias(type=nil, &blk) # TODO TODO TODO
+    if blk
+      T::Private::Types::TypeAlias.new(blk)
```

Comment:
Nice! We'll need to describe the change to open-source folks.
Are we also writing a rubocop for automatic conversion? Can we opensource it?